src/api/modules/auth/Guards/userGuard.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class UserGuard implements CanActivate {
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+      const request = context.switchToHttp().getRequest();
+      const user = request.user
```

Comment:
this variable is redundant

Reply:
sir please how do you mean. I don't understand

Reply:
the variable user is not really needed there

you can use `request.user` directly.

in your code, if you are not using a variable In more than one place, then don't make a variable, use that value directly